project/Dependencies.scala

```diff
@@ -6,7 +6,7 @@ object Dependencies {
   // WARNING: Please Scala update versions in PluginCross.scala too
   val scala212 = "2.12.19"
   val scala213 = "2.13.12"
-  val scala3 = "3.3.1"
+  val scala3 = "3.5.1-RC1-bin-20240628-1efbb92-NIGHTLY"
```

Comment:
Using a nightly version because of https://github.com/scala/scala3/pull/20592, and we need the Zinc products to find the auto plugins.

Eventually we should use `3.3.5` or `3.5.1`.